kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -98,6 +100,7 @@ class DeltaTableSchemaEvolutionSuite extends DeltaTableWriteSuiteBase with Colum
 
       val structType = table.getLatestSnapshot(engine).getSchema
       assertColumnMapping(structType.get("a"), 1)
```

Comment:
it would be good to assert there that "c" is not in fact in the schema?

Also, I'm not sure if we should have a test case to ensure the next column ID assigned is 3?

Reply:
I agree on probably all counts -- but trying to scope this PR to just fixing the bug and testing the bug fix @amogh-jahagirdar maybe include this in the tests-follow ups?

Reply:
Yeah I should fix this test ....

Reply:
More broadly I can cleanup the vast majority of the tests to just compare the new and actual schemas via equality. That'll compare metadata as well. No need to unpack every field and compare